### PR TITLE
soc_evcc: add pin for fiat

### PR DESCRIPTION
**Pull request by ts78**
Fiat requires an app pin for SoC access, so add optional field to enter that pin and only use it if car type is fiat.

### CODE REVIEW

**Comment by ts78** [comment]
I have not yet tested this code, therefore I converted it to draft.

**Comment by ts78** [comment]
I have tested and fixed functionality from openWB point of view (UI changes, config changes, etc). I have not tested fiat soc request due to missing fiat credentials - the command being triggered for fiat SoC is:

```
/var/www/html/openWB/modules/soc_evcc/../soc_evcc/soc fiat --user abc@mail.com --password abc --vin 123456abc --token 123456abc --pin abcpin123
```

So it should work as intended.

**Comment by Bucky2k** [comment]
Looks good on a first glance. I see the prompt for PIN is suppressed for the other car types. But is it submitted always to the binary in case for non-fiat prompts?

**Comment by ts78** [comment]
> Looks good on a first glance. I see the prompt for PIN is suppressed for the other car types. But is it submitted always to the binary in case for non-fiat prompts?

The value entered at settings page is always stored in `openwb.conf`, but the field to enter PIN is only shown if `fiat` has been selected as car type. In the call to evcc soc binary, the option `--pin` is only added if car type is `fiat`, see https://github.com/snaptec/openWB/pull/2631/files#diff-f858327b01ac958689ba2a347876562cb85592d51dacb83ad152c66bb4180d51L75-R81

**Comment by Bucky2k** [comment]
Perfect, then it should work I guess. Sorry, I am on travel and only had a quick glance. Now up to snaptec to adopt it. Then a Fiat nightly user can test

File: modules/soc_evcc/main.sh [hunk 1]
```diff
@@ -72,7 +74,11 @@ incrementTimer(){
 getAndWriteSoc(){
 	openwbDebugLog ${DMOD} 0 "Lp$CHARGEPOINT: Requesting SoC"
 	echo 0 > $soctimerfile
-	answer=$($MODULEDIR/../soc_evcc/soc $fztype --user "$username" --password "$password" --vin "$vin" --token "$token" 2>&1)
+	options="${fztype} --user ${username} --password ${password} --vin ${vin} --token ${token}"
+	if [ "${fztype}" == "fiat" ]; then
+		options="${options} --pin ${pin}"
+	fi
+	answer=$($MODULEDIR/../soc_evcc/soc $options 2>&1)
```
Comment (benderl) [inline modules/soc_evcc/main.sh:81]: Es ist etwas sinnfrei, in das übergeordnete Verzeichnis zu wechseln, um dann wieder zurückzugeben.
```suggestion
	answer=$($MODULEDIR/soc $options 2>&1)
```

File: modules/soc_evcc/main.sh [hunk 2]
```diff
@@ -86,7 +92,11 @@ getAndWriteSoc(){
 }
 wakeupCar(){
 	openwbDebugLog ${DMOD} 0 "Lp$CHARGEPOINT: Wakeup car"
-	answer=$($MODULEDIR/../soc_evcc/soc $fztype --user "$username" --password "$password" --vin "$vin" --token "$token" --action wakeup 2>&1)
+	options="${fztype} --user ${username} --password ${password} --vin ${vin} --token ${token}"
+	if [ "${fztype}" == "fiat" ]; then
+		options="${options} --pin ${pin}"
+	fi
+	answer=$($MODULEDIR/../soc_evcc/soc $options --action wakeup 2>&1)
```
Comment (benderl) [inline modules/soc_evcc/main.sh:99]:
```suggestion
	answer=$($MODULEDIR/soc $options --action wakeup 2>&1)
```

**Review by benderl — CHANGES_REQUESTED** [review]

File: modules/soc_evcc/main.sh (hunk 1 above)
Comment (ts78) [reply modules/soc_evcc/main.sh:81]: Das war im Original schon so, wurde also offensichtlich vorher in einem Review nicht bemängelt:
```
answer=$($MODULEDIR/../soc_evcc/soc $fztype --user "$username" --password "$password" --vin "$vin" --token "$token" 2>&1)
```

Aber ja, das sehe ich auch so. Habe die Änderungsvorschläge übernommen, Danke!

**Review by benderl — APPROVED** [review]

**Pull request merged**